apps/backend/api/services/Search/util.js

```diff
@@ -75,14 +75,17 @@ export const filterAndSortPosts = curry((opts, q) => {
     })
   }
 
-  if (afterTime) {
+  if (afterTime && beforeTime) {
+    q.where(q2 =>
+      q2.where('posts.start_time', '<', beforeTime)
+      .andWhere('posts.end_time', '>=', afterTime)
+    )
+  } else if (afterTime) {
     q.where(q2 =>
       q2.where('posts.start_time', '>=', afterTime)
       .orWhere('posts.end_time', '>=', afterTime)
     )
-  }
-
-  if (beforeTime) {
+  } else if (beforeTime) {
     q.where(q2 =>
       q2.where('posts.start_time', '<', beforeTime)
       .andWhere('posts.end_time', '<', beforeTime)
```

Comment:
this one is a judgement call, and you made a different judgement above when both are set. does beforeTime mean the whole event needs to be before that time? I said yes, but you said no. I actually think you are right again. so we can definitely simplify this whole thing into two separate condtions using the ones you used.